spec/models/person_spec.rb

```diff
@@ -34,5 +34,53 @@
         end
       )
     end
+
+    # it "association reify error behaviour = :warn" do
+    #   #::PaperTrail.config.association_reify_error_behaviour = :warn
+
+    #   person = Person.create(name: "Frank")
+    #   thing = Thing.create(name: "BMW 325")
+    #   thing = Thing.create(name: "BMX 1.0")
+
+    #   person.thing = thing
+    #   person.thing = thing
+    #   person.update_attributes(name: "Steve")
+
+    #   thing.update_attributes(name: "BMW 330")
+    #   thing.update_attributes(name: "BMX 2.0")
+    #   person.update_attributes(name: "Peter")
+
+    #   expect(person.reload.versions.length).to(eq(3))
+
+    #   expect(person.versions.first.logger).to(
+    #     receive(:warn).with(/Unable to reify has_one association/).twice
+    #   )
+
+    #   person.reload.versions.second.reify(has_one: true)
+    # end
+
+    # it "association reify error behaviour = :ignore" do
+    #   #::PaperTrail.config.association_reify_error_behaviour = :ignore
+
+    #   person = Person.create(name: "Frank")
+    #   thing = Thing.create(name: "BMW 325")
+    #   thing = Thing.create(name: "BMX 1.0")
+
+    #   person.thing = thing
+    #   person.thing = thing
+    #   person.update_attributes(name: "Steve")
+
+    #   thing.update_attributes(name: "BMW 330")
+    #   thing.update_attributes(name: "BMX 2.0")
+    #   person.update_attributes(name: "Peter")
+
+    #   expect(person.reload.versions.length).to(eq(3))
+
+    #   expect(person.versions.first.logger).to_not(
+    #     receive(:warn).with(/Unable to reify has_one association/)
+    #   )
+
+    #   person.reload.versions.second.reify(has_one: true)
+    # end
```

Comment:
If you're having trouble writing this type of test, I'd be happy with testing `load_version` in isolation, even using mocks. I'd be happier with a test that used real database records, but for now an isolated test is OK with me.